### PR TITLE
Add the service catalog item ordering via REST.

### PR DESCRIPTION
**Pull request by mfalesni**
Also adds a new feature, `pytest.wait_for` wrapper. Usage in the newly written test. @psav has already seen it, @seandst ?

{{pytest: cfme/tests/services/test_service_catalogs.py -v --long-running -k rest}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 4339a3fa5e88dc430237a59b4c37ed8421131299:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit e752e7d274473216c478d9f2a169a4fc03333171:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6affc016e24e1cc74bd324feaa027f74e933623b:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 0ca6d7cecc2f499a30674aab9e2e71f46fc885b9:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 04175ac511c129e9947d100737c90734255c2822:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by mfalesni** [comment]
@psav , @seandst ^ This fails only on rhev, according to logs due to rhev's slowness.

**Comment by mfalesni** [comment]
@psav ping?

**Comment by mfalesni** [comment]
@psav I also extended `wait_for` to process `timedelta` and strings with specific format to make the timeout more literal.

**Comment by dajoRH** [comment]
Lint report for commit 7741ae2a1c32ee0a255e2db4bd8303294b2d4354:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 44a793e513f7b8fed09d41ff90149f0fcbbeb675:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 0d8e06b1fa79f77062d0395f5388c1d70e53ea29:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by mfalesni** [comment]
@psav ^?

File: fixtures/wait.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+"""This module contains interation between pytest and :py:mod:`utils.wait`."""
```
Comment (psav) [inline fixtures/wait.py]: spelling?

**Comment by dajoRH** [comment]
Lint report for commit f480f2423a7e72758d6b75c38382bf1bd3334c90:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by seandst** [comment]
@mfalesni, I poked a little bit at `parsedatetime` and I'm not sure it works as well as we need/want, though I might be making the requirements a little too broad.

First, I'm sad that a module called `parsedatetime` returns time tuples. Second, notice how putting multiple fragments breaks the time parsing:

```
17 >>> c.parse('3d')
   17: 
(time.struct_time(tm_year=2015, tm_mon=8, tm_mday=6, tm_hour=16, tm_min=43, tm_sec=57, tm_wday=3, tm_yday=218, tm_isdst=-1),
 1)

18 >>> c.parse('3d1h')
   18: 
(time.struct_time(tm_year=2015, tm_mon=8, tm_mday=3, tm_hour=16, tm_min=44, tm_sec=1, tm_wday=0, tm_yday=215, tm_isdst=1),
 0)

19 >>> c.parse('1h')
   19: 
(time.struct_time(tm_year=2015, tm_mon=8, tm_mday=3, tm_hour=17, tm_min=44, tm_sec=33, tm_wday=0, tm_yday=215, tm_isdst=-1),
 2)
```

`3d`(prompt 17) adds three days, as you'd expect. `1h` (prompt 19) adds 1 hour, again, just like you'd expect. But `3d1h` (prompt 18) seems to completely give up. Neither the `3d` nor the `1h` is evaluated. The [`parse` docs] (https://bear.im/code/parsedatetime/docs/parsedatetime.Calendar-class.html#parse) lead me to believe that this usage should work, but I could be misreading them or simply looking at the wrong method.

Let me know what you think.

**Comment by mfalesni** [comment]
@seandst I have nothing against putting the space there (as I documented it in the `wait_for`) like `"3m 10s"`, which works.

**Comment by dajoRH** [comment]
Lint report for commit 264e3b6e325d7f63e20dc4d4cfdfb7f8b5b7d828:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Pull request merged**